openedx/core/djangoapps/programs/utils.py

```diff
@@ -577,3 +582,37 @@ def _collect_instructors(self, course_run):
             self.instructors.update(
                 {instructor.get('name'): instructor for instructor in course_instructors.get('instructors', [])}
             )
+
+    def _collect_one_click_purchase_eligibility_data(self):
+        applicable_seat_types = self.data['applicable_seat_types']
+        is_learner_eligible_for_one_click_purchase = self.data['is_program_eligible_for_one_click_purchase']
+        skus = []
+        if is_learner_eligible_for_one_click_purchase:
+            for course in self.data['courses']:
+                is_learner_eligible_for_one_click_purchase = not any(
+                    course_run['is_enrolled'] for course_run in course['course_runs']
+                )
+                if is_learner_eligible_for_one_click_purchase:
+                    for course_run in course['course_runs']:
+                        for seat in course_run['seats']:
+                            if seat['type'] in applicable_seat_types:
+                                skus.append(seat['sku'])
+                else:
+                    skus = []
+                    break
+
+        if skus:
+            try:
+                User = get_user_model()
+                service_user = User.objects.get(username=settings.ECOMMERCE_SERVICE_WORKER_USERNAME)
+                api = ecommerce_api_client(service_user)
+                # Make an API call to calculate the discounted price
+                results = api.baskets.calculate.get(sku=skus)
+                self.data['discounted_price'] = results.get('total_incl_tax')
+            except SlumberBaseException:
+                log.exception('Failed to get discount price for following product SKUs: %s ', ', '.join(skus))
+
+        self.data.update({
+            'basket_page_url': self.ecommerce_service.checkout_page_url(*skus) if skus else '#courses',
```

Comment:
Code smell! Why is there an if statement here? Your data model definitely should *not* care about an element ID from the view.